### PR TITLE
Strict types + typehinting for AppData

### PR DESCRIPTION
**Pull request by rullzer**
Noticed while touching code that uses the AppData stuff ;)

### CODE REVIEW

**Review by juliusknorr — DISMISSED** [review]

**Comment by juliushaertl** [comment]
Tests are failing.

**Comment by rullzer** [comment]
will take care

File: apps/theming/tests/IconBuilderTest.php [hunk 1]
```diff
@@ -93,7 +93,7 @@ public function dataRenderAppIcon() {
 	 * @param $color
 	 * @param $file
 	 */
-	public function testRenderAppIcon($app, $color, $file) {
+	public function XtestRenderAppIcon($app, $color, $file) {
```
Comment (MorrisJobke) [inline apps/theming/tests/IconBuilderTest.php]: Test excluded?

**Comment by MorrisJobke** [comment]
Also phan throws a lot of errors. :/

File: apps/theming/tests/IconBuilderTest.php (hunk 1 above)
Comment (rullzer) [reply apps/theming/tests/IconBuilderTest.php]: Ah let me fix

**Comment by rullzer** [comment]
Yes the phan errors are thrown on all branches. I bet something got added to master

File: lib/private/Files/SimpleFS/SimpleFile.php [hunk 2]
```diff
@@ -54,7 +55,7 @@ public function getName() {
 	 *
 	 * @return int
 	 */
-	public function getSize() {
+	public function getSize(): int {
```
Comment (nickvergessen) [inline lib/private/Files/SimpleFS/SimpleFile.php]: breaks for 32bit with bigger files?

Comment (rullzer) [reply lib/private/Files/SimpleFS/SimpleFile.php]: Well the callers already expect an int. Also looking down the code it seems it really is an int.

**Comment by rullzer** [comment]
Lets do this later

**Pull request closed without merging**